### PR TITLE
doc for Automated Device Configuration

### PR DESCRIPTION
**Pull request by jkralik**
## Summary by CodeRabbit

- **Documentation**
  - Updated various configuration documents to include new settings for message size limits and authorization enhancements for the gRPC API.
  - Introduced OpenTelemetry exporter configuration settings across multiple services for improved observability.
  - Added a new section for "Automated Device Configuration," describing how the Snippet Service helps ensure accurate device setup and automatically applies predefined configurations in response to events.

### CODE REVIEW

File: content/en/docs/configuration/certificate-authority.md [hunk 1]
```diff
@@ -33,11 +33,13 @@ A configuration template is available on [certificate-authority/config.yaml](htt
 
 ### gRPC API
 
-gRPC API of the Certificate Authority service as defined [here](https://github.com/plgd-dev/hub/blob/main/certificate-authority/pb/service_grpc.pb.go#L19).
+gRPC API of the Certificate Authority service as defined [here](https://github.com/plgd-dev/hub/blob/main/certificate-authority/pb/service.proto#L34).
 
 | Property | Type | Description | Default |
 | ---------- | -------- | -------------- | ------- |
 | `apis.grpc.address` | string | `Listen specification <host>:<port> for grpc client connection.` | `"0.0.0.0:9100"` |
+| `apis.grpc.sendMsgSize` | int32 | `Set the max message size in bytes the server can send. 0 means 2147483647` | `4194304` |
```
Comment (Danielius1922) [inline content/en/docs/configuration/certificate-authority.md:41]: 0 means default value right, so shouldn't the default be 2147483647?

File: content/en/docs/services/snippet-service/features.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+description: 'What features does the Snippet Service provide?'
+date: '2024-07-12'
+categories: [snippet-service, automation, configuration]
+keywords: [device-configuration, snippet-service]
+weight: 2
+---
+
+## Description
+
+This feature allows users to pre-configure devices by specifying desired configurations that are automatically applied when a ResourceChanged event occurs. The newly introduced `snippet service` is responsible for managing this process.
```
Comment (Danielius1922) [inline content/en/docs/services/snippet-service/features.md:12]: For consistency we should probably enclose all ResourceChanged words in "``", so all occurences are visually the same

File: content/en/docs/configuration/snippet-service.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+| `clients.storage.mongoDB.tls.certFile` | string | `File path to certificate in PEM format.` | `""` |
+| `clients.storage.mongoDB.tls.useSystemCAPool` | bool | `If true, use system certification pool.` | `false` |
+
+### Resource Aggregate Client
+
+Resource Aggregate client is responsible for updating resources in the device when conditions are satisfied or on-demand by the user.
+
+| Property | Type | Description | Default |
+| ---------- | -------- | -------------- | ------- |
+| `clients.resourceAggregate.cleanUpExpiredUpdates` | string | `Cron expression to clean up expired updates.` | `"0 * * * *"` |
```
Comment (Danielius1922) [inline content/en/docs/configuration/snippet-service.md:113]: move to storage client

File: content/en/docs/services/snippet-service/features.md (hunk 2 above)
Comment (jkralik) [reply content/en/docs/services/snippet-service/features.md]: fixed

File: content/en/docs/configuration/snippet-service.md (hunk 3 above)
Comment (jkralik) [reply content/en/docs/configuration/snippet-service.md]: moved

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**